File: python/백준/집합과 맵/1794 듣보잡.py
import sys

input = sys.stdin.readline

N, M = map(int, input().split())

never_heard = set()
never_seen = set()
for i in range(N):
    never_heard.add(input().rstrip())

for i in range(M):
    never_seen.add(input().rstrip())

never_heard_seen = never_heard & never_seen
print(len(never_heard_seen))

for name in sorted(never_heard_seen):
    print(name)

File: cpp/백준/집합과 맵/1794 듣보잡.cpp
#include <iostream>
#include <string>
#include <set>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N, M;
    cin >> N >> M;

    set<string> never_heard;
    set<string> never_seen;

    for (int i = 0; i < N; ++i) {
        string name;
        cin >> name;
        never_heard.insert(name);
    }
    for (int i = 0; i < M; ++i) {
        string name;
        cin >> name;
        never_seen.insert(name);
    }

    vector<string> never_heard_seen;
    for (const auto& name : never_heard) {
        if (never_seen.find(name) != never_seen.end()) {
            never_heard_seen.push_back(name);
        }
    }

    cout << never_heard_seen.size() << "\n";
    for (const auto& name : never_heard_seen) {
        cout << name << "\n";
    }
    return 0;
}
